Test mode

quintessence opened the pull request:
Added test mode, refactored handling of streaming msgs into a function. Updated README.

quintessence committed:
pretty print, key filtering

quintessence committed:
while -> for, fixed fall through

quintessence committed:
removed temp variable

quintessence committed:
Created test mode and corresponding test file

jhunt commented on logemongo:
```diff
@@ -17,8 +20,10 @@ my %OPTIONS = (
 	password => 'admin',
 	include  => [],
 	exclude  => [],
+	filter   => [],
```
This won't be used, since the option below is `filters|filter`

jhunt commented on logemongo:
```diff
@@ -30,23 +35,83 @@ GetOptions(\%OPTIONS, qw/
-my @EXCLUDES = map { qr/$_/ } @{$OPTIONS{exclude}};
+sub handle {
+	my ($message) = @_;
+	my ($logdata, $payload, %msghash);
```
It's not great form to name the `%thing` with "hash" on the end -- the `%` shows that it's a hash.

jhunt commented on logemongo:
```diff
@@ -30,23 +35,83 @@ GetOptions(\%OPTIONS, qw/
+	while ($logdata =~ m/$qr/) {
+		$msghash{$1}=$2;
+		$logdata =~ s/$qr//;
+	}
```
After a substitution, the `$1` and `$2` back refs will still be useful.  You don't need the `qr//` here.

Consider:

```
my $log = "key=value other=value";
while ($log =~ s/(\w*?)=([^ ]*)//) {
    print "found '$1' = '$2'\n";
}
```

jhunt commented on logemongo:
```diff
@@ -30,23 +35,83 @@ GetOptions(\%OPTIONS, qw/
+	if (@EXCLUDES) {
+		for my $re (@EXCLUDES) {
+			if ($payload =~ $re) {
+				return 1;
```
Should this be `return 0` - i.e. we skipped because it matched and exclude?

jhunt commented on logemongo:
```diff
@@ -30,23 +35,83 @@ GetOptions(\%OPTIONS, qw/
+			}
+		}
+		return 0;
+	}
```
Does this still stream all logs when nothing is included, excluded or filtered?

jhunt commented on logemongo:
```diff
@@ -30,23 +35,83 @@ GetOptions(\%OPTIONS, qw/
 my $URL = "https://$OPTIONS{host}:$OPTIONS{port}/ws";
 
-print "Targeting sawmill at '$URL'.\n\nDepending on your configuration it may take a moment for log\nmessages to appear. Please be patient.\n\n";
+$LOGEMONGO_TEST ? print "Testing with test log file.\n" : print "Targeting sawmill at '$URL'.\n\nDepending on your configuration it may take a moment for log\nmessages to appear. Please be patient.\n\n";
```
I would move these prints to inside and after the `if` block, respectively.

jhunt reviewed (changes requested):
Fix the merge conflicts, and tweak the things that need tweaking.  Otherwise looks good!

quintessence committed:
Fixed handle func when streaming, also:
- removed `filter => []`
- s/msghash/msgfield/
- fixed regex
- adjusted print statement

quintessence committed:
resolved merge conflicts

jhunt commented on README.md:
```diff
@@ -31,3 +32,6 @@ logemongo -H ${host} -P ${port} -u ${username} -p ${password}
 By default, `logemongo` uses `10.244.73.2:443` as the `host:port` and `admin:admin` for the `user:pass` since these are the values that are used when running a local Sawmill on BOSH lite. If you are testing against a Sawmill running on BOSH lite you will also want to disable SSL with the `--no-ssl` flag as BOSH lite uses self-signed certificates. For a complete list options and syntax, use the `- ...
+=======
+CLI for Sawmill BOSH release.
+>>>>>>> master
```
Missed a conflict or two.

jhunt commented on logemongo:
```diff
@@ -53,25 +53,25 @@ sub DEBUG {
-	$msghash{'payload'} = $payload;
+	($msgfield{timestamp}, $msgfield{uuid}) =  split ' ', $message;
+
+  while ($log =~ s/(\w*?)="(.*?)"//) {
```
Check your whitespace here.

jhunt reviewed (changes requested):
Nothing major.

quintessence committed:
Resolved conflict in README

quintessence committed:
Fixed whitespaces
